lib/SparqlEndpointFetcher.ts

```diff
@@ -243,9 +242,18 @@ export class SparqlEndpointFetcher {
     if (!options?.ignoreBody && httpResponse.body) {
       // Wrap WhatWG readable stream into a Node.js readable stream
       // If the body already is a Node.js stream (in the case of node-fetch), don't do explicit conversion.
-      responseStream = <NodeJS.ReadableStream>(
-        isStream(httpResponse.body) ? httpResponse.body : new ReadableWebToNodeStream(httpResponse.body)
-      );
+      if (isStream(httpResponse.body)) {
+        responseStream = <NodeJS.ReadableStream> <unknown> httpResponse.body;
+      } else {
+        const httpResponseBodyReader = httpResponse.body.getReader();
```

Comment:
So the [Readable.fromWeb](https://nodejs.org/api/stream.html#streamreadablefromwebreadablestream-options) does not exist in readable-stream?

If not, to what extent does the code below correspond to Node's Readable.fromWeb impl? Just to know how stable and tested the code below is.

Reply:
Yes, the `Readable.fromWeb` does not exist in `readable-stream`. They have a `from` that supports `AsyncInterable` and something else, but not the response body directly as-is.

The code does approximately the same thing as Node's own implementation [here](https://github.com/nodejs/node/blob/0b676736a0e9ab4939c195a516aa7e82fcd839aa/lib/internal/webstreams/adapters.js#L545) in the `read` method, except it does not destroy the stream upon error, just emits the error as an event.

It seems to work the same way as `Readable.fromWeb` with regards to successful and failed streams, based on testing with the SPARQL benchmark runner and the unit test. The new unit test I added makes sure the errors can be handled through the error events in the converted `Readable` stream, when they were previously uncaught.

I do not know how to further test this, so any tips would be welcome.

Reply:
Could you also manually test this in a browser build of comunica targeting a SPARQL endpoint?
The easiest would probably be to link this within the comunica jquery widget, and then sending some queries to sparql endpoints.

Reply:
I will try to reimplement the entire `fromWeb` function using Node's as reference. No idea how it will work.

Reply:
Can't we just stick with `readable-web-to-node-stream` then? We know that works. No need to re-invent the wheel, unless you see a good reason?

Reply:
The `readable-web-to-node-stream` library does not handle errors at all, and it crashes the whole program with an uncaught error exception when the web stream errors out, such as when the server terminates the connection or the undici body timeout is reached. That is what also causes the SPARQL benchmark runner to crash when the SPARQL endpoint crashes or otherwise closes the connection.

Reply:
I could look into fixing the fork of `readable-web-to-node-stream` next week, I guess. This uncaught error has already taken two weeks or more, so spending a day or two on fixing that should not be that bad.

Reply:
> The readable-web-to-node-stream library does not handle errors at all, and it crashes the whole program with an uncaught error exception when the web stream errors out

Ok, that's a good point indeed.

Fixing the fork then sounds like a good idea. Perhaps you could even create your own fork (maybe we even want it under the comunica namespace)?
If I remember correctly, `readable-web-to-node-stream` is used in another place in comunica, so having that reusable fork would be good.

Reply:
There is now [the library](https://github.com/comunica/readable-from-web.js) under Comunica that handles this conversion without crashing everything, and I have updated the code in this PR to use that, instead.